server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -253,6 +255,10 @@ public ThreadPool(
             new ScalingExecutorBuilder(Names.TRANSLOG_TRANSFER, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5))
         );
         builders.put(Names.TRANSLOG_SYNC, new FixedExecutorBuilder(settings, Names.TRANSLOG_SYNC, allocatedProcessors * 4, 10000));
+        builders.put(
+            Names.TRANSLOG_REMOTE_PURGE,
+            new ScalingExecutorBuilder(Names.TRANSLOG_REMOTE_PURGE, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5))
```

Comment:
Does `Scaling` threadpool have a backing queue? If yes, what is the size? If not, then what happens if all threads are busy?

Reply:
It uses a variant of `LinkedTransferQueue` - https://github.com/opensearch-project/OpenSearch/blob/83dccad312faa7d9350ffa6672e51d02d36ad06f/server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchExecutors.java#L132-L155

It is unbounded in size. If all threads are busy, it would be waiting in the queue.